### PR TITLE
Flag vector counts

### PR DESCRIPTION
**Pull request by JosephTremoulet**
Mark immediates which came from the Vector<T>.Count intrinsic, and loops that have such as their limit.

### CODE REVIEW

**Comment by JosephTremoulet** [comment]
@dotnet/jit-contrib PTAL.  This is a step on the way to #7843.

**Comment by erozenfeld** [comment]
LGTM

File: src/jit/optimizer.cpp [hunk 1]
```diff
@@ -822,6 +822,10 @@ bool Compiler::optCheckIterInLoopTest(
     if (limitOp->gtOper == GT_CNS_INT)
     {
         optLoopTable[loopInd].lpFlags |= LPFLG_CONST_LIMIT;
+        if (limitOp->gtFlags & GTF_ICON_SIMD_COUNT)
```
Comment (sivarv) [inline src/jit/optimizer.cpp]: CC: (limitOp->gtFlags & GTF_ICON_SIMD_COUNT) != 0

**Comment by sivarv** [comment]
Looks good.

**Comment by CarolEidt** [comment]
:shipit:

**Pull request merged**